### ISSUE TITLE
Monospace is detected as variable-width

### ISSUE THREAD

**Issue by es20490446e** [lxqt/qterminal#1088]
If in a terminal I launch `qterminal 1.4.0` it says:

> Using a variable-width font in the terminal.  This may cause performance degradation and display/alignment errors.

But the selected font is `monospace`.

**Comment by tsujan** [comment]
The font may be advertised to be monospace, but Qt may find out that it really isn't: https://doc.qt.io/qt-6/qfontinfo.html#fixedPitch, QTerminal just follows Qt.

In such cases, it might be a good idea to report the problem to the creator(s) of the font.

**Comment by tsujan** [comment]
What about DejaVu Sans Mono? I don't see any message with it here.

**Comment by tsujan** [comment]
Oh, create the file `~/.config/fontconfig/conf.d/0-mono.conf` and add this to it:

```xml
<?xml version='1.0'?>
<!DOCTYPE fontconfig SYSTEM 'fonts.dtd'>
<fontconfig>
 <!-- To fix monospace -->
 <alias>
  <family>monospace</family>
  <prefer>
   <family>DejaVu Sans Mono</family>
  </prefer>
 </alias>
 <alias>
  <family>serif</family>
  <prefer>
   <family>DejaVu Serif</family>
  </prefer>
 </alias>
 <alias>
  <family>sans-serif</family>
  <prefer>
   <family>DejaVu Sans</family>
  </prefer>
 </alias>
</fontconfig>
```

Then try again.

**Comment by tsujan** [comment]
@yan12125, shouldn't this info be added to the wiki? All Qt apps that need monospace depend on it.

**Comment by yan12125** [comment]
Feel free to go ahead and update the wiki.

**Comment by tsujan** [comment]
Done.

**Comment by yan12125** [comment]
This may need some more investigation. It is reported that the aforementioned fonfconfig file is causing an issue [1][2].

Reopening as a reminder for myself.

[1] https://gitlab.archlinux.org/archlinux/packaging/packages/ttf-dejavu/-/issues/1
[2] https://gitlab.archlinux.org/archlinux/packaging/packages/ttf-dejavu/-/issues/2

**Comment by tsujan** [comment]
> The issue seems to be with the font itself:

When the problem happened here last year, I found a relation to Noto. The default monospace changed (it was DejaVu for years, and I was completely satisfied with it) to a font whose monospace variant didn't have a distinguishable bold weight in FeatherPad's syntax highlighting, and that happened when Noto and `fontconfig` were both upgraded. However, I thought `fontconfig` might be the culprit.

> So the wiki shall not suggest that work-around

The user can get rid of these problems by setting monospace, and the user `fontconfig` exists for a reason. So, what's proposed in the wiki is completely valid.

**Comment by tsujan** [comment]
> Monospaced is already set at /etc/fonts/conf.d/60-latin.conf.

Yes, but users may want to set it to a *fixed* font of their choice, especially when the system default changes badly — as was the case with Arch.

**Comment by yan12125** [comment]
> Global confs go to:

Looks like there is indeed an issue with out-of-box fontconfig, but it does not really make sense to make your personal preference system-wide, let alone making it distribution-wide like https://gitlab.archlinux.org/archlinux/packaging/packages/ttf-dejavu/-/issues/1. Please think twice before proposing changes that can affect many users.

Meanwhile, it updated https://github.com/lxqt/qterminal/wiki#how-to-set-my-preferred-font-for-monospace and make it clearer.
